### PR TITLE
CATL-1358: Make "Coming Soon" block configurable

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of this PR, the "Show Coming soon block" settings has been added, which controls the visibility of `Next Milestone`, `Next Activity` and `Calendar` blocks in Case Details Summary tab.

## Before
2020-05-13 at 7 53 PM

## After
**Settings**
2020-05-13 at 7 54 PM

**Case Summary Page**
(When "Show Coming soon block" is set to NO)
2020-05-13 at 7 52 PM

## Technical Details
**Configurable Blocks in Case Summary Tabs**
A new provider `CaseDetailsSummaryBlocks` has been created. Using this we can configure the the Blocks in Case details summary tab. 
For now we are adding the `Next Milestone`, `Next Activity` and `Calendar` using a `config` file when `Show Coming Soon Case Summary Block` settings is true.

**Moving of markups and placeholders into Directives**
* As part of this PR, a lot of markup files has been moved to their own directive. This makes the markup more readable.
Example
```html
<div
  ng-include="'~/civicase/case/details/directives/case-details-header.html'">   
</div>

<!-- vs -->

<civicase-case-details-header></civicase-case-details-header>
```

* Also, previously all the placeholders for different Case Details tabs, were placed under `ang/civicase/case/details/directives/placeholder/` folder. But now that we are moving into the direction of making blocks configurable, its the best time to move the placeholders to their respective directives. 
To achieve the same, all the placeholder files have been deleted, and they are moved into their newly created individual directives.
Inside `case-details.directive.js`, two new functions(`isPlaceHolderVisible` and `isMainContentVisible`) are created which decides when to display/hide placeholders.
This way all the small components are responsible to show/hide their placeholders.

**`civicase-include-replace` directive**
This new directive has been created. The `ngInclude` directive always creates an extra element, but sometimes we need the template to replace the `ng-include` element. This directive is used in such cases.

### CODE REVIEW

File: ang/civicase/case/details/configs/add-case-summary-blocks.config.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    var caseSummaryBlocks = [
+      {
+        templateUrl: '~/civicase/case/details/summary-tab/case-details-summary-next-milestone.html',
+        weight: 0
+      }, {
+        templateUrl: '~/civicase/case/details/summary-tab/case-details-summary-next-non-milestone-activity.html',
+        weight: 0
+      }, {
+        templateUrl: '~/civicase/case/details/summary-tab/case-details-summary-calendar.html',
+        weight: 0
```
Comment (reneolivo) [inline ang/civicase/case/details/configs/add-case-summary-blocks.config.js:20]: why not use weight 0, 1, 2?

File: ang/civicase/case/details/summary-tab/case-details-summary-next-non-milestone-activity.html [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      crm-popup-form-success="pushCaseData($data.civicase_reload[0])"
+      refresh-callback="refresh"
+      mode="big"
+      type="'activity'"
+      case="item">
+    </div>
+  </div>
+
+  <!-- Placeholder -->
+  <div ng-if="isPlaceHolderVisible()" class="civicase__loading-placeholder__oneline" style="width: 10em; margin-bottom: 10px;"></div>
```
Comment (reneolivo) [inline ang/civicase/case/details/summary-tab/case-details-summary-next-non-milestone-activity.html:17]: on the one hand I'm happy that the HTML parts are stored into easy to find folders, but on the other I find having the actual content and the placeholder content in a single file hard to follow.

Is it possible to have something like this for everyone:

```html
<div class="panel civicase__summary-tab-tile civicase__panel-transparent-header civicase__summary-tab-tile--responsive">
  <div ng-if="isMainContentVisible()" class="panel-heading">
    <h3 class="panel-title">Next Activity</h3>
  </div>
  <div ng-if="isMainContentVisible()" class="panel-body">
    <div
      case-activity-card="item.nextActivityNotMilestone"
      crm-popup-form-success="pushCaseData($data.civicase_reload[0])"
      refresh-callback="refresh"
      mode="big"
      type="'activity'"
      case="item">
    </div>
  </div>

  <div
    ng-if="isPlaceHolderVisible()"
    ng-include="'some-path-next-to-this-file/name-of-file__placeholder.html'">
  </div>
</div>
```

File: ang/civicase-base/providers/case-details-summary-blocks.provider.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      caseDetailsSummaryBlocks = caseDetailsSummaryBlocks.concat(itemsConfig);
+    }
+  });
+})(angular, CRM.$, CRM._);
+
+/**
+ * @typedef {object} ButtonConfig
+ * @property {string} templateUrl
+ * @property {number} weight
+ */
```
[the remark this answers is not in the document]
Comment (deb1990) [reply ang/civicase-base/providers/case-details-summary-blocks.provider.js:40]: There was a small issue, it should be renamed to `BlockConfig`. Changed.

File: ang/civicase/case/details/configs/add-case-summary-blocks.config.js (hunk 1 above)
Comment (deb1990) [reply ang/civicase/case/details/configs/add-case-summary-blocks.config.js:20]: Oops, that was a mistake. Used 0, 5, 10, so that items can be added in the middle.

File: ang/civicase/case/details/summary-tab/case-details-summary-next-non-milestone-activity.html (hunk 2 above)
Comment (deb1990) [reply ang/civicase/case/details/summary-tab/case-details-summary-next-non-milestone-activity.html:17]: @reneolivo I initially tried this, but that way a new html element gets created because of `ng-include` and it changes the html tree, which messes up with the styles. Now I have created `civicaseIncludeReplace` to fix this particular issue, but still I was a bit reluctant to use this all over the place, because there are so many file changes in this PR, that in case something goes wrong, it will take much more time to everything.
Can we keep it as it is for now? as we have a good start, and we can build on top of this in future, where we will change few files at a time, so that its easier to fix in case of regression.
Thats why I added the `<!-- placeholder -->` comment in each file so that its easier to understand.

Comment (reneolivo) [reply ang/civicase/case/details/summary-tab/case-details-summary-next-non-milestone-activity.html:17]: Agreed. Let's try to move them as we touch the files. This is way better than the previous implementation anyways.
